### ISSUE TITLE
enable-https fails with certbot error

### ISSUE THREAD

**Issue by Pertel** [nextcloud/nextcloud-snap#492]
Installed v 12.0.5snap2 on a debian 9 server, and am getting an error when trying to enable https:

I'm running `/snap/bin/nextcloud.enable-https lets-encrypt` and getting the following error after inputting email and domain:

`certbot: error: unrecognized arguments: --max-log-backups 0`

I'm a bit confused, since I've looked in the enable-https script [here](https://github.com/nextcloud/nextcloud-snap/blob/12.0.5snap2/src/https/scripts/enable-https#L107), and it does not seem to use that argument when calling the certbot.

I believe that the `max-log-backups` argument is only supported in certbot v 0.17 and forward as seen in their [changelog](https://github.com/certbot/certbot/blob/master/CHANGELOG.md#0170---2017-08-02)

I seems that the installed certbot is is v 0.14.1 when looking in the [setup.py](https://github.com/nextcloud/nextcloud-snap/blob/12.0.5snap2/src/https/setup.py#L8), so perhaps I'm missing something in my understanding.

It looks like the certbot is somehow called with --max-log-backups, or perhaps some other argument is being interpreted incorrectly as that argument.

**Comment by kyrofa** [comment]
Wha... ? Yeah, we don't use that anywhere. Let me see if I can duplicate.

**Comment by kyrofa** [comment]
Hmm... are snaps confined on Debian? Do you have cerbot in your PATH by any chance? Run `which certbot` from the terminal where you ran enable-https. I bet you'll find a newer one there, which the snap may be using since I think it's accessible on Debian, but then it calls into the OLDER one in the snap, which barfs.

**Comment by Pertel** [comment]
I did have another certbot installed via apt, but have since uninstalled it again. But it still might be related somehow. I tried again and made sure to uninstall the apt package, and `which certbot` does not return anything. The error still occurs sadly.

But it sounds like it could be related to the apt package.

**Comment by kyrofa** [comment]
Did you run `apt autoremove` as well? Might have left python bits behind.

**Comment by Pertel** [comment]
I ran `apt-get remove python-certbot-apache` and then `apt-get autoremove`.

**Comment by Pertel** [comment]
Hmm. Removing the nextcloud snap and installing it again did not fix the problem.

**Comment by Pertel** [comment]
Ah, perhaps it is because the version of certbot being installed is greater than 0.14.1.

Looking at the [documentation for setuptools](https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires) and the `install_requires`, it looks like it just specifies a minimum version to be installed. Looking at [setup.py](https://github.com/nextcloud/nextcloud-snap/blob/12.0.5snap2/src/https/setup.py#L8) it indeed uses ìnstall_requires` to specify a certbot version.

I did try to find which certbot version was actuall installed, but i had no luck running the certbot binary without some sort of error. Perhaps someone more knowledgeable in snaps can figure out how to check which certbot version is actually used in the currently installed snap.

**Comment by Pertel** [comment]
I got it working!

As suspected, it was due to a previously installed certbot on the machine, in the form of a certbot apt package. Somehow the snap command was picking it up from its PATH, even though it was not in the PATH of my shell. Must have been too tired to find it properly yesterday.

I guess that snaps are not completely confined from existing packages, at least not in this case. 
Thanks for the help
